### ISSUE TITLE
Can not add new device: ERROR RPC error 504: Error: DB error (code 504)

### ISSUE THREAD

**Issue by c-jonua** [keybase/client#8395]
I'm was download the keybase app on work computer; next I was try to login them. App was generates a secret phrase for me and was shown a message, what I need to add a new device in my authorized app. When I came home, I was trying to add a new devise as described abow, I was specify the next command:

    keybase device add

next I was select the "(1) Desktop or laptop" options and type my verification code. Then I was reseived next messages:

    ✔ Verification code received.
    ▶ ERROR RPC error 504: Error: DB error (code 504)

my log id: 9896a1000f6d539ce300cd1c

**Comment by maxtaco** [comment]
I think you get this error if your first attempt fails, and then you try to reuse the QR code a second time (or third time). Does it work if you quit the app and then try again? 

Thanks for your patience and also the good bug report.

Cc: @cjb and @chrisnojima

**Comment by cjb** [comment]
@maxtaco I dunno, DB error sounds to me like it's coming from the API server..

**Comment by cjb** [comment]
Oh, I suppose this could have been during Aurora migration downtime today?

**Comment by maxtaco** [comment]
I found the error on the API server, it comes from a primary key violation of trying to insert a 1st message into a KEX session for a second time.  The primary key is on (sessionID, packetSequenceNumber).

**Comment by maxtaco** [comment]
(It's sanitized so as not to reveal the SQL running on the server, but the server logs show the full error).

**Comment by c-jonua** [comment]
Small clarification: On the work computer I made **two** attempts to login in app (accordingly, it was generated two codes). On home notebook firstly I was plased first code, gets the error as described abow and then was try again with second code. And received the same error message again

Cc: @maxtaco and @cjb

**Comment by maxtaco** [comment]
Did you restart both your phone and your desktop app?  My guess is that you only restarted the desktop ap.

**Comment by c-jonua** [comment]
I dont using phone: both app installed on desktop. I restarts the home machine. The work machine I can starts in a few hours. Work machine is off at the moment.

**Comment by maxtaco** [comment]
Oh! Yeah, we don't really support this mode of operation.  Both devices need to be online at the same time.  It doesn't work if one is offline and one is online.  Use a "paper key" for that if you don't want to use an iPhone app.

**Comment by c-jonua** [comment]
Oh, really.. Okay, I will trying to use paperkey.
When I read the keybase docs I was not unerstanding this. May be you guys can expands the docs, can be added more information about use cases paperkeys & new devices?

**Comment by c-jonua** [comment]
And, error message can be more informativelly :)

**Comment by gtirloni** [comment]
I had this issue today on a computer where I dual boot Windows and Linux. Installed Keybase on Windows, setup everything, installed on Linux, got the code, rebooted into Windows, error 504!

**Comment by maxtaco** [comment]
@gtirloni let me see if I understand properly.  You didn't have both computers running at the same time, you were turning one off, and then turning the other on?  That's definitely not how we intend the protocol to work, but it will help us a bunch to tell us exactly what you tried. Thank you

**Comment by gtirloni** [comment]
@maxtaco yes, but it's a single computer that dual boots Windows and Linux. Understood now it won't work that way. Thanks!

**Comment by maxtaco** [comment]
We have to work on better error messaging and errors for people who do this. My sense is you are not the first person who went down this path...

**Comment by neekro** [comment]
Hi guys, 

I'm also getting this error while trying to add a new computer device. I added a different device (mobile) prior to this and it went well, but not with this one. 

First I tried to add it using the new mobile using QR code. After pointing at the QG code, the app took more than a minute and nothing happened so I killed it and retry again. However, since then I always got this error. Even after I tried to use a different device for adding the new one or while not using QR code but rewrote the sentences instead. Always the same error:

**There was an error provisioning** 
RPC error 504: Error: DB error Error code 504: DB_UPDATE_ERROR in method keybase.1.device.deviceAdd

**Comment by makarid** [comment]
Hello, i am getting the same error when i am trying to add a phone device

**Comment by tinanwa** [comment]
I cannot add this device telling me not seen. How do I go about it.
